.github/workflows/publish.yml

```diff
@@ -94,7 +94,11 @@ jobs:
 
             - name: Set ui-common version
               working-directory: packages/ui-common
-              run: yarn version set ${{ steps.compute.outputs.version }}
+              env:
+                  VERSION: ${{ steps.compute.outputs.version }}
+              run: |
+                  node -e "const fs=require('fs'); const p=JSON.parse(fs.readFileSync('package.json','utf8')); p.version=process.env.VERSION; fs.writeFileSync('package.json', JSON.stringify(p,null,2)+'\n');"
```

Comment:
Ok, I was able to test this fix, which does indeed work. But this line is very hard to read and understand. Can you pull this logic out into a small special purpose script (and place it in build_scripts). Then call the script in the yaml here. This also ensures the script is subject to shellcheck, so be sure to create the script with that in mind, and document the script clearly for later users.